ci/check-go-version.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# Make sure the minor Go version that we are running matches the version specified in go.mod
+goVersion=$(go version | # looks something like "go version go1.20.1 darwin/amd64"
+  awk '{print $3}' |     # get the 3rd word -> "go1.20.1"
+  sed "s/go//" |         # remove the "go" part -> "1.20.1"
+  sed 's/\.[^.]*$//')    # remove the last period to end -> "1.20"
```

Comment:
also there was a slight bug in this, i can push a fix in a sec

Reply:
thanks!